src/network/networkd-network.c

```diff
@@ -228,8 +231,11 @@ int network_load_one(Manager *manager, const char *filename) {
                               "CAN\0",
                               config_item_perf_lookup, network_network_gperf_lookup,
                               CONFIG_PARSE_WARN, network);
-        if (r < 0)
+        if (r < 0) {
+                /* Unset manager here. Otherwise, LIST_REMOVE() in network_free() fails. */
```

Comment:
This definitely looks better!

But how about initializing `*network = (Network) { ... }` without setting `.manager = ...` there and set it here after the check instead?

```c
if (r < 0)
    return r;
/* Set manager here, it's safe to do so at this point. */
network->manager = manager;
```

Or, even better (if that works), just before (or right after) the `LIST_PREPEND` a few lines down, since then if the code before it is modified to return on error, the problem with `network_free` won't be reintroduced...

Reply:
It might be possible, but I am lazy about checking whether all config_parse_xxx() work without manager. So, I chose the current form. And just I guess some functions require manager.

Anyway, I will check that later. Thanks.

Reply:
Yeah, I found `config_parse_netdev()` requires manager is set.